Refactored examples and added github example with README

savitaashture opened the pull request:
# Changes
Refactored examples and added github example along with README
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

dibyom commented on examples/github/README.md:
```diff
@@ -0,0 +1,44 @@
+
+   The response status code should be `201 Created`
+   
+   [`HMAC`](https://www.freeformatter.com/hmac-generator.html) tool used to create X-Hub-Signature
```
Thanks for adding this! Might be overkill but we could mention that the "string" is the body and the secretKey is from the example secret?

dibyom commented on examples/github/README.md:
````diff
@@ -0,0 +1,44 @@
+1. You should see a new PipelineRun that got created:
+
+   ```shell script
+   kubectl get pipelinerun
````
Again, if we used our own pipeline/taskrun here, it would be easier to track what got created

dibyom commented on examples/github/binding.yaml:
```diff
@@ -0,0 +1 @@
+../role-resources/triggerbinding-roles/binding.yaml
```
As I mentioned in #578 , I'd prefer adding the binding/template/pipeline/secret in this folder in the same file as it makes it much clearer what is going on (just look at this one file vs looking at 6-7 different files). I know this leads to some duplication but for examples I'm ok with having some repetition in order to help with clarity

dibyom commented:
Thanks for addressing the comments. Just two things:
1. Could we put the role/binding/serviceaccount file in one file (say, role.yaml)?
2. Can we squash the commits into one?
Otherwise LGTM!

savitaashture commented:
> Thanks for addressing the comments. Just two things:
> 
> 1. Could we put the role/binding/serviceaccount file in one file (say, role.yaml)?
> 2. Can we squash the commits into one?
>    Otherwise LGTM!

Thank you
1. Modified based on the suggestion
2. squashed commits

I have a question here is if we squash commits we actually loose the history of review comments fix :thinking: 
I know this is small PR but in case of big feature its actually make sense to keep the history of commits so that it will be easy for the reviewers to see actual fix.

dibyom commented:
> I have a question here is if we squash commits we actually loose the history of review comments fix 🤔
I know this is small PR but in case of big feature its actually make sense to keep the history of commits so that it will be easy for the reviewers to see actual fix.

Yeah, unfortunately GitHub code review does not handle diffs across commits well. And it is indeed helpful during review to see what comments are addressed. However, I think squashing still makes sense. Otherwise, the master branch gets cluttered with a bunch of small commits which makes it hard to rollback/cherry-pick (as well as figure out exactly what happened).

dibyom commented:
/approve
/lgtm
